### PR TITLE
Ruby IceLocal cleanup

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR cleans up what used to be the Local Slice generated code in Ruby.

In particular, it moves the remaining files from IceLocal to Ice (next to the generated code).

### CODE REVIEW

File: ruby/ruby/.gitignore [hunk 1]
```diff
@@ -1,5 +1,17 @@
 Glacier2/*
-Ice/*
+Ice/BuiltinSequences.rb
```
Comment (bernardnormier) [inline ruby/ruby/.gitignore:2]: Must list all the generated files here.

File: ruby/ruby/Glacier2.rb [hunk 2]
```diff
@@ -2,6 +2,6 @@
 # Copyright (c) ZeroC, Inc. All rights reserved.
 #
 
-require 'Glacier2/Router'
-require 'Glacier2/PermissionsVerifier'
-require 'Glacier2/Metrics'
+require_relative 'Glacier2/Router'
```
Comment (bernardnormier) [inline ruby/ruby/Glacier2.rb:5]: We should consider _not_ shipping this generated code. That is, if a user wants to create a Ruby client to Glacier2/IceGrid/IceStorm, this user can just compile the corresponding Slice files. Not for this PR.

File: ruby/ruby/Ice/CompressBatch.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Copyright (c) ZeroC, Inc.
+
+module Ice
+
+    class CompressBatch
```
Comment (bernardnormier) [inline ruby/ruby/Ice/CompressBatch.rb:5]: The only remaining former local types are enums and local exceptions.

File: ruby/src/IceRuby/Connection.cpp [hunk 4]
```diff
@@ -316,10 +316,10 @@ IceRuby::initConnection(VALUE iceModule)
     //
     // Instance members.
     //
+    rb_define_attr(_connectionInfoClass, "underlying", 1, 0);
```
Comment (bernardnormier) [inline ruby/src/IceRuby/Connection.cpp:319]: We define ConnectionInfo/EndpointInfo in C++. However, these definitions were buggy and "fixed" (augmented) by the code generated from local Slice...

File: ruby/src/IceRuby/Types.cpp [hunk 5]
```diff
@@ -3169,7 +3149,6 @@ IceRuby::initTypes(VALUE iceModule)
     rb_define_module_function(iceModule, "__defineDictionary", CAST_METHOD(IceRuby_defineDictionary), 3);
     rb_define_module_function(iceModule, "__declareProxy", CAST_METHOD(IceRuby_declareProxy), 1);
     rb_define_module_function(iceModule, "__declareClass", CAST_METHOD(IceRuby_declareClass), 1);
```
Comment (bernardnormier) [inline ruby/src/IceRuby/Types.cpp:3151]: I guess we have declare and define class for forward-declaration purposes? Is this really necessary in Ruby?

Comment (pepone) [reply ruby/src/IceRuby/Types.cpp:3151]: I think it is with the current code, if the forward declare type is used in an operation we pass the type to the method that defines the operation.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

**Pull request merged**